Feature 29 statusausgabe

JulianRetzbach opened the pull request.

JulianRetzbach committed:
Update Muehle_GUI.ipynb

JulianRetzbach committed:
Update Muehle_GUI.ipynb

JulianRetzbach committed:
Update Muehle_GUI.ipynb

luisafuss commented on Muehle_GUI.ipynb:
```diff
@@ -220,6 +233,98 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def on_button_clicked(b):\n",
+    "    status.pieces= [[9,9],[          \n",
+    "                    [0,0,0,0,0,0,0,0], \n",
+    "                    [0,0,0,0,0,0,0,0], \n",
+    "                    [0,0,0,0,0,0,0,0]  \n",
+    "                   ]]\n",
+    "    status.current_player = 1\n",
+    "    status.mills = set()\n",
+    "    status.number_stones_to_remove = 0\n",
+    "    status.selected_dot_old = (None, None)\n",
+    "    status.selected_dot_new = (None, None)\n",
+    "    status.move_or_jump = False\n",
+    "    status.game_over = False\n",
```
Ist das so geschickt, wenn wir das doppelt haben? Wenn wir eine Variable hinzufügen, müssen wir sie an beiden Stellen ändern... Ich weiß aber gerade nicht, wie man das schöner löst

JulianRetzbach replied:
was genau meinst mit andere Variante? Gibts etwa mehrere?

luisafuss replied:
Was für eine Variante? wir haben die Angaben nur genau so auch in der Klasse status() stehen... Das heißt, wenn wir dort eine Variable ergänzen, müssen wir sie auch in on_button_clicked() ergänzen. Das hat man schnell vergessen. Außerdem sagt der Methoden-Name aktuell nichts über die Button-Funktion aus, die diese Methode aufruft. Ist das Absicht, oder wäre etwas wie "on_restart_button_clicked()" sinnvoller?

luisafuss commented on Muehle_GUI.ipynb:
```diff
@@ -34,7 +43,7 @@
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 2,
```
Vor dem mergen nochmal den execution_count zurücksetzen

luisafuss commented on Muehle_GUI.ipynb:
```diff
@@ -220,6 +233,98 @@
   },
+  {
+   "source": [
+    "## Initialisierung und Funktionen der Status-Widgets und des Restar-Button"
```
```suggestion
    "## Initialisierung und Funktionen der Status-Widgets und des Restart-Buttons"
```

luisafuss commented on Muehle_GUI.ipynb:
```diff
@@ -220,6 +233,98 @@
+    "                min=0,\n",
+    "                description='Spieler 1',\n",
+    "                bar_style='info', # 'success', 'info', 'warning', 'danger' or ''\n",
+    "                orientaiton='horizontal'\n",
```
Ist `orientation` nicht falsch geschrieben?
```suggestion
    "                orientation='horizontal'\n",
```

JulianRetzbach replied:
ist aus der doku

luisafuss replied:
![image](https://user-images.githubusercontent.com/54306427/104033846-61282380-51d0-11eb-937e-0786239232ce.png)

luisafuss commented on Muehle_GUI.ipynb:
```diff
@@ -220,6 +233,98 @@
+    "                min=0,\n",
+    "                description='Spieler 2',\n",
+    "                bar_style='', # 'success', 'info', 'warning', 'danger' or ''\n",
+    "                orientaiton='horizontal'\n",
```
Hier dann auch

luisafuss commented on Muehle_GUI.ipynb:
```diff
@@ -220,6 +233,98 @@
+    "                max=9,\n",
+    "                min=0,\n",
+    "                description='Spieler 1',\n",
+    "                bar_style='info', # 'success', 'info', 'warning', 'danger' or ''\n",
```
Vielleicht kann man die auch an die Steinfarbe der Spieler anpassen

luisafuss replied:
Und richtig cool wäre es, wenn die Statusanzeige rechts neben dem Spielfeld ist. Dann hat man das schöner in einem Blickfeld.

JulianRetzbach replied:
Steinfarben hab ich jetzt so gut wie möglich angepasst.

luisafuss commented on Muehle_GUI.ipynb:
```diff
@@ -220,6 +233,98 @@
+    "    turn_label.value='Spieler '+ str(status.current_player)+ ' ist dran.'\n",
+    "    status_label.value = 'Status: '+'Spieler in Phase '+ str(get_player_phase(status.pieces, status.current_player))+'.'\n",
+    "    if(status.pieces[0][0] == status.pieces[0][1] == 0):\n",
+    "        pieces_status_label = widgets.Label(value='Verbleibende Steine:')\n",
```
Das Label ist bei mir bei "setzbare Steine" geblieben...

JulianRetzbach replied:
ist behoben

luisafuss commented on Muehle_GUI.ipynb:
```diff
@@ -220,6 +233,98 @@
+    "    status_label.value = 'Status: '+'Spieler in Phase '+ str(get_player_phase(status.pieces, status.current_player))+'.'\n",
+    "    if(status.pieces[0][0] == status.pieces[0][1] == 0):\n",
+    "        pieces_status_label = widgets.Label(value='Verbleibende Steine:')\n",
+    "        stones_player_one.value = len(player_pieces(status.pieces, 1))\n",
+    "        stones_player_two.value = len(player_pieces(status.pieces, 2))\n",
```
Es gibt eine Funktion `count_player_pieces()`

JulianRetzbach replied:
werd ich ersetzen

luisafuss commented on Muehle_GUI.ipynb:
```diff
@@ -264,13 +375,14 @@
     "        board[2].fill_style = 'black'\n",
     "        board[2].font = '18px serif'\n",
     "        board[2].fill_text('Spieler '+ str(status.current_player), BOARD_SIZE*0.4, BOARD_SIZE*0.45)\n",
-    "        board[2].fill_text('ist dran.', BOARD_SIZE*0.4, BOARD_SIZE*0.55)\n",
+    "        board[2].fill_text('ist dran.', BOARD_SIZE*0.4, BOARD_SIZE*0.55)  \n",
```
Das kann ja dann eigentlich weg

JulianRetzbach replied:
stimmt, wollte ich zum Vergleich aber noch drin lassen

luisafuss commented on Muehle_GUI.ipynb:
```diff
@@ -333,19 +445,107 @@
+    }
+   ],
+   "source": [
+    "display(restart_button)\n"
```
Wenn ich da drauf klicke, wird nochmal die ganze Statusanzeige darunter eingeblendet

JulianRetzbach replied:
ist jetzt nichtmehr vorhanden

luisafuss commented:
closes #29

JulianRetzbach committed:
Update Muehle_GUI.ipynb

JulianRetzbach committed:
Update Muehle_GUI.ipynb

JulianRetzbach committed:
widgets side by side

Anordnung der Widgets jetzt nebeneinander

JulianRetzbach committed:
Bugfix

pieces_status_label zeigt jetzt den richtigen Wert: Verbleibende Steine oder Setzbare Steine.

JulianRetzbach committed:
redundante Funktionen löschen
